library/core/src/intrinsics/simd.rs

```diff
@@ -243,12 +243,13 @@ extern "platform-intrinsic" {
     ///
     /// `T` must be a vector.
     ///
-    /// `U` must be a vector of pointers to the element type of `T`, with the same length as `T`.
+    /// `U` must be a pointer to the element type of `T`
     ///
     /// `V` must be a vector of integers with the same length as `T` (but any element size).
     ///
     /// For each element, if the corresponding value in `mask` is `!0`, read the corresponding
-    /// pointer from `ptr`.
+    /// pointer offset from `ptr`.
+    /// The first element is loaded from `ptr`, the second from `ptr.wrapping_offset(1)` and so on.
```

Comment:
i'd expect it to use `offset`, not `wrapping_offset`

Reply:
The entire point of this is to be able to mask off loads that would be out-of-bounds. If it used `offset`, that would defeat the purpose.

Reply:
> The entire point of this is to be able to mask off loads that would be out-of-bounds. If it used `offset`, that would defeat the purpose.

no, because `offset` is only called for those indexes that are not masked off -- sort-of. LLVM IR semantics are quite lacking here:
> The ‘llvm.masked.load’ intrinsic is designed for conditional reading of selected vector elements in a single IR operation. It is useful for targets that support vector masked loads and allows vectorizing predicated basic blocks on these targets. Other targets may support this intrinsic differently, for example by lowering it into a sequence of branches that guard scalar load operations. The result of this operation is equivalent to a regular vector load instruction followed by a ‘select’ between the loaded and the passthru values, predicated on the same mask. However, using this intrinsic prevents exceptions on memory access to masked-off lanes.

Reply:
I'm proposing to clarify the LangRef: https://github.com/llvm/llvm-project/pull/82469.